features/grpc/exporter/src/test/java/org/opennms/features/grpc/exporter/NmsInventoryServiceSyncIT.java

```diff
@@ -0,0 +1,598 @@
+public class NmsInventoryServiceSyncIT implements TemporaryDatabaseAware<MockDatabase> {
+
+    private static final int PORT = 50051;
+    private static final String TENANT_ID = "opennms-prime";
+    private static final String HOST_NAME = "localhost:" + PORT;
+
+    private Server server;
+    private SpogGrpcClient client;
+
+    @Autowired
+    private DatabasePopulator databasePopulator;
+
+    @Autowired
+    private HwEntityDao hwEntityDao;
+
+    @Autowired
+    private NodeDao nodeDao;
+
+    @Autowired
+    private AlarmDao alarmDao;
+
+    @Autowired
+    private AlarmDao eventDao;
+
+    @Autowired
+    private MockEventIpcManager eventdIpcMgr;
+
+    private int _id = 1;
+    private MockDatabase mockDatabase;
+
+    private NmsInventoryServiceSyncImpl grpcServerService = null;
+    private CompletableFuture<Boolean> serverResponseFuture = null;
+
+    @Before
+    public void setUp() throws Exception {
+        eventdIpcMgr.setEventWriter(mockDatabase);
+        // create data for node
+        populateData();
+        serverResponseFuture = new CompletableFuture<>();
+        grpcServerService = new NmsInventoryServiceSyncImpl(serverResponseFuture);
+        // configure server and client
+        initializeAndStartServer();
+        initializeAndStartClient();
+    }
+
+    private void populateData() {
+
+        databasePopulator.addExtension(new DatabasePopulator.Extension<HwEntityDao>() {
+            @Override
+            public DatabasePopulator.DaoSupport<HwEntityDao> getDaoSupport() {
+                return new DatabasePopulator.DaoSupport<HwEntityDao>(HwEntityDao.class, hwEntityDao);
+            }
+
+            @Override
+            public void onPopulate(DatabasePopulator populator, HwEntityDao dao) {
+                OnmsNode node = databasePopulator.getNode1();
+                dao.flush();
+            }
+
+            @Override
+            public void onShutdown(DatabasePopulator populator, HwEntityDao dao) {
+                for (OnmsHwEntity entity : dao.findAll()) {
+                    dao.delete(entity);
+                }
+            }
+        });
+        databasePopulator.populateDatabase();
+    }
+
+    private void initializeAndStartServer() throws IOException {
+        server = ServerBuilder.forPort(PORT)
+                .addService(grpcServerService)
+                .build()
+                .start(); // Restart
+    }
+
+    private void initializeAndStartClient() throws SSLException {
+        client = new SpogGrpcClient(HOST_NAME,
+                null,
+                false,
+                new GrpcHeaderInterceptor(TENANT_ID)
+        );
+        client.start();
+    }
+
+    @Test
+    public void testClientRecoveryAfterServerStopAndRestart() throws Exception {
+        // Step 1: Wait for the channel to be ready and set up the initial inventory update
+        waitForChannelToBeReady();
```

Comment:
Ideally, you don't need explicitly check for channel to be ready.   Using awaitility, you can wait till it succeeds.